src/forms/dispense-form.component.tsx

```diff
@@ -44,101 +41,82 @@ const DispenseForm: React.FC<DispenseFormProps> = ({
     if (!isSubmitting) {
       setIsSubmitting(true);
       const abortController = new AbortController();
-      medicationDispensesPayload.map((dispenseRequest) => {
-        saveMedicationDispense(dispenseRequest, abortController).then(
-          ({ status }) => {
-            if (status === 201 || status === 200) {
-              closeOverlay();
-              mutate();
-              showToast({
-                critical: true,
-                kind: "success",
-                description: t(
-                  "medicationListUpdated",
-                  "Medication dispense list has been updated."
-                ),
-                title: t(
-                  mode === "enter"
-                    ? "medicationDispensed"
-                    : "medicationDispenseUpdated",
-                  mode === "enter"
-                    ? "Medication successfully dispensed."
-                    : "Dispense record successfully updated."
-                ),
-              });
-            }
-          },
-          (error) => {
-            showNotification({
+      saveMedicationDispense(medicationDispensePayload, abortController).then(
+        ({ status }) => {
+          if (status === 201 || status === 200) {
+            closeOverlay();
+            mutate();
+            showToast({
+              critical: true,
+              kind: "success",
+              description: t(
+                "medicationListUpdated",
+                "Medication dispense list has been updated."
+              ),
               title: t(
                 mode === "enter"
-                  ? "medicationDispenseError"
-                  : "medicationDispenseUpdatedError",
+                  ? "medicationDispensed"
+                  : "medicationDispenseUpdated",
                 mode === "enter"
-                  ? "Error dispensing medication."
-                  : "Error updating dispense record"
+                  ? "Medication successfully dispensed."
+                  : "Dispense record successfully updated."
               ),
-              kind: "error",
-              critical: true,
-              description: error?.message,
             });
-            setIsSubmitting(false);
           }
-        );
-      });
-    }
-  };
-
-  // updates the medication dispense referenced by the specified index, this function passed on to the card to allow updating a specific element
```

Comment:
we can get rid of this method entirely now that we don't have to individually update different dispense elements in an array